pkg/util/shoot_clients.go

```diff
@@ -69,7 +69,7 @@ func NewShootClients(c client.Client, clientset kubernetes.Interface, gardenerCl
 // NewClientForShoot returns the rest config and the client for the given shoot namespace.
 func NewClientForShoot(ctx context.Context, c client.Client, namespace string, opts client.Options) (*rest.Config, client.Client, error) {
 	gardenerSecret := &corev1.Secret{}
-	if err := c.Get(ctx, kutil.Key(namespace, v1beta1constants.SecretNameGardener), gardenerSecret); err != nil {
+	if err := c.Get(ctx, kutil.Key(namespace, v1beta1constants.SecretNameGardenerInternal), gardenerSecret); err != nil {
```

Comment:
Can we have a fallback to the `SecretNameGardener` in case the `SecretNameGardenerInternal` is not found?

Reply:
good point.